File: src/types/Post.ts
import { User } from "./User";

export type Post = {
  id?:number;
  likes?:number;
  views?:number;
  reposts?:number;
  replies?: number;
  repost?: Post;
  text?: string;
  mediaPath?: string;
  ownerId?: number;
  owner?: User;
  datePosted?: Date;
  rootId: number;
  parentId: number;
  grandparent?: Post;
  parent?: Post;
  isLiked?: boolean;
  hasReposted?: boolean;
}

File: TwitterClone/Models/PostModels/Post.cs
namespace TwitterClone.Models.PostModels
{
    public class Post
    {
        public int Id {  get; set; }

        public int? RootId { get; set; }
        public int? ParentId { get; set; } 
        public int Likes { get; set; }
        public int Views { get; set; }
        public int Reposts { get; set; }
        public int Replies { get; set; }
        public Post? Repost { get; set; }
        public string? Text { get; set; }
        public string? MediaPath { get; set; }
        public required int OwnerId { get; set; }
        public User? Owner { get; set; }
        public DateTime? DatePosted { get; set; }
    }
}
